### PR TITLE
fix: refresh nonce after signing

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

File: runtime/src/rpc.rs [hunk 1]
```diff
@@ -70,12 +72,30 @@ impl PolkaBtcProvider {
...
+        )
+        .await
+        .map_or_else(|_| u32::MIN, |info| info.nonce);
+
+        account_nonce = std::cmp::max(system_nonce, account_nonce);
+
+        let mut cloned_signer = signer.clone();
+        cloned_signer.set_nonce(account_nonce);
+
+        account_nonce = account_nonce.wrapping_add(1);
```
Comment (sander2) [inline runtime/src/rpc.rs:94]: Did you check if wrapping will work? If this is supported, I am surprised that `increment_nonce` is not doing this already

File: runtime/src/rpc.rs [hunk 2]
```diff
@@ -70,12 +72,30 @@ impl PolkaBtcProvider {
...
-        // return the current value, increment afterwards
-        let ret = signer.clone();
-        signer.increment_nonce();
-        ret
+        let mut account_nonce = self.nonce.load(Ordering::SeqCst);
+
+        // For getting the nonce, use latest, possibly non-finalized block.
+        // TODO: we might want to wait until the latest block is actually finalized
+        // query account info in order to get the nonce value used for communication
+        let system_nonce = crate::frame_system::AccountStoreExt::account(
```
Comment (sander2) [inline runtime/src/rpc.rs:81]: This is significant overhead, can we not get away with not doing this, and instead doing this only if a transaction fails due to the nonce being outdated? Presumably this shouldn't happen very often (only when the user is making transactions from the UI, right?)

Comment (gregdhill) [reply runtime/src/rpc.rs]: I don't think think the overhead will be too high, but in any case I've adapted this PR to only refresh on error.

**Review by sander2 — APPROVED** [review]

File: runtime/src/rpc.rs (hunk 2 above)
Comment (sander2) [reply runtime/src/rpc.rs]: Cool, thanks

**Pull request merged**